src-ghc/Pact/Server/PactService.hs

```diff
@@ -64,36 +65,52 @@ initPactService CommandConfig {..} loggers = do
 
 applyCmd :: Logger -> Maybe EntityName -> PactDbEnv p ->
             GasModel -> Word64 -> Int64 -> ExecutionMode -> Command a ->
-            ProcessedCommand PublicMeta ParsedCode -> IO CommandResult
-applyCmd _ _ _ _ _ _ _ cmd (ProcFail s) = return $ jsonResult Nothing (cmdToRequestKey cmd) (Gas 0) s
+            ProcessedCommand PublicMeta ParsedCode -> IO (CommandResult [TxLog Value])
+applyCmd _ _ _ _ _ _ _ cmd (ProcFail s) =
+  return $ resultFailure Nothing (cmdToRequestKey cmd) (Gas 0) $
+  PactError TxFailure def def . viaShow $ s
 applyCmd logger conf dbv gasModel bh bt exMode _ (ProcSucc cmd) = do
   let pubMeta = _pMeta $ _cmdPayload cmd
       (ParsedDecimal gasPrice) = _pmGasPrice pubMeta
       gasEnv = GasEnv (fromIntegral $ _pmGasLimit pubMeta) (GasPrice gasPrice) gasModel
 
   let pd = PublicData pubMeta bh bt
 
-  r <- tryAny $ runCommand (CommandEnv conf exMode dbv logger gasEnv pd) $ runPayload cmd
+  r <- catchesPactError $ runCommand (CommandEnv conf exMode dbv logger gasEnv pd) $ runPayload cmd
   case r of
     Right cr -> do
       logLog logger "DEBUG" $ "success for requestKey: " ++ show (cmdToRequestKey cmd)
       return cr
     Left e -> do
       logLog logger "ERROR" $ "tx failure for requestKey: " ++ show (cmdToRequestKey cmd) ++ ": " ++ show e
-      return $ jsonResult Nothing (cmdToRequestKey cmd) (Gas 0) $
-               CommandError "Command execution failed" (Just $ show e)
+      -- Linda TODO
+      return $ resultFailure Nothing (cmdToRequestKey cmd) (Gas 0) e
 
-jsonResult :: ToJSON a => Maybe TxId -> RequestKey -> Gas -> a -> CommandResult
-jsonResult tx cmd gas a = CommandResult cmd tx (toJSON a) gas
 
+resultFailure :: Maybe TxId ->
+                 RequestKey ->
+                 Gas ->
+                 PactError ->
+                 CommandResult [TxLog Value]
+resultFailure tx cmd gas a = CommandResult cmd tx (PactResult . Left $ a) gas Nothing Nothing Nothing
```

Comment:
we're always going to fail with `Gas 0`, so we can probably factor that out of `applyCmd`